js/hitbtc3.js

```diff
@@ -1497,6 +1498,67 @@ module.exports = class hitbtc3 extends Exchange {
         };
     }
 
+    async fetchFundingRateHistory (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = undefined;
+        const request = {
+        // all arguments are optional
+        // 'symbols': Comma separated list of symbol codes,
+        // 'sort': 'DESC' or 'ASC'
+        // 'from': 'Datetime or Number',
+        // 'till': 'Datetime or Number',
+        // 'limit': 100,
+        // 'offset': 0,
+        };
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request['symbols'] = market['id'];
+        }
+        if (since !== undefined) {
+            request['from'] = since;
+        }
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const response = await this.publicGetPublicFuturesHistoryFunding (this.extend (request, params));
+        //
+        //    {
+        //        "BTCUSDT_PERP": [
+        //            {
+        //                "timestamp": "2021-07-29T16:00:00.271Z",
+        //                "funding_rate": "0.0001",
+        //                "avg_premium_index": "0.000061858585213222",
+        //                "next_funding_time": "2021-07-30T00:00:00.000Z",
+        //                "interest_rate": "0.0001"
+        //            },
+        //            ...
+        //        ],
+        //        ...
+        //    }
+        //
+        const contracts = Object.keys (response);
+        const rates = [];
+        for (let i = 0; i < contracts.length; i++) {
+            const marketId = contracts[i];
+            const market = this.safeMarket (marketId);
+            const entry = response[marketId];
+            for (let i = 0; i < entry.length; i++) {
+                const symbol = this.safeSymbol (market['symbol']);
+                const fundingRate = this.safeNumber (entry[i], 'funding_rate');
```

Comment:
instead of accessing `entry[i]` more than once, save `entry[i]` to a variable and then access the variable, you might realize line 1550 is wrong also